Migrate topic log compaction (#402)
aiven/devportal: docs/products/kafka/concepts/partition-segments.rst

Partition segments
==================

> Apache Kafka divides topics partition data into **segment** files (with ``.log`` suffix) stored on the file system. Each segment file is named using the offset of the first message (a.k.a. **base offset**) contained. E.g. the segment file ``04.log`` contains the message with offset ``4`` as first entry. 

Reviewer: ```suggestion
Apache Kafka divides topics partition data into **segment** files (with ``.log`` suffix) stored on the file system. Each segment file is named using the offset of the first message (a.k.a. **base offset**) contained. For example, the segment file ``04.log`` contains the message with offset ``4`` as first entry. 
```
both are right, I just think looks better a non abbreviated version here as it is the beginning of a sentence but this is IMHO

The last segment in the partition is called the **active segment** and it is the only segment to which new messages are appended to.

> In the example above you a topic partition is divided into multiple segments, with ``06.log`` being the active one:

Reviewer: ```suggestion
In the example above the topic partition is divided into multiple segments, with ``06.log`` being the active one:
```
do we need to specify which topic partition is? I think it was meant to be **your** in any case.

**01.log**

.. list-table::
  :header-rows: 1
  :stub-columns: 1
  :align: left

  * - Offset
    - Key
    - Value
  * - 1
    - 1001 
    - 4 Privet Dr
  * - 2
    - 1002
    - 221B Baker Street
  * - 3
    - 1003
    - Milkman Road

**04.log**

.. list-table::
  :header-rows: 1
  :stub-columns: 1
  :align: left

  * - Offset
    - Key
    - Value
  * - 4
    - 1002
    - 21 Jump St
  * - 5
    - 1001
    - Paper St

**06.log (active segment)**

.. list-table::
  :header-rows: 1
  :stub-columns: 1
  :align: left

  * - Offset
    - Key
    - Value
  * - 6
    - 1001
    - Paper Road 21

> When the segment file reaches a certain size of age, Apache Kafka will create a new segment file. This can be controlled by the following settings:

Reviewer: off topic: but can we know which size of age is that? I am not super familiar with kafka but do we know if `size of age` is a particular term on kafka? size x age in general so this why I am curious :)

Reviewer 2: it was meant to be `size OR age`,thanks

> -  ``segment.bytes`` : create a new segment when current segment becomes greater than this size. This setting can be set during topic creation and defaults to 1GB.

Reviewer: As we describing of all those segments as a segment. Would make sense to ignore their inner name and just treat as a third person?
```suggestion
-  ``segment.bytes`` : creates a new segment when current segment becomes greater than this size. This setting can be set during topic creation and defaults to 1GB.
```
